joint_teapot/workers/gitea.py

```diff
@@ -375,6 +378,32 @@ def archive_all_repos(self) -> None:
                 self.org_name, repo.name, body={"archived": True}
             )
 
+    def get_all_teams(
+        self,
+    ) -> List[Dict[str, Union[str, List[str]]]]:
+        ret: List[Dict[str, Union[str, List[str]]]] = []
+        try:
+            teams_raw = self.organization_api.org_list_teams(self.org_name)
+        except ApiException as e:
+            logger.error(f"Failed to get teams from organization {self.org_name}: {e}")
+            exit(1)
+        for team_raw in teams_raw:
+            if team_raw.name == "Owners":
+                continue
+            team_id = team_raw.id
+            try:
+                members = [
+                    m.login.lower()
+                    for m in self.organization_api.org_list_team_members(team_id)
+                ]
+            except ApiException as e:
+                logger.warning(
+                    f"Failed to get members of team {team_id} in {self.org_name}: {e}"
+                )
+                continue
+            ret.append({"name": team_raw.name, "members": members})
```

Comment:
`{team_raw.name, members}` is enough for now.

Reply:
I don't think this a good idea as it goes against the model of dictionaries. I would rather use tuples if you insist on removing the field names.

Reply:
What do you mean by "the model of dictionaries"?

Reply:
Dictionaries are hash tables, i.e. they are used to look up values according to keys. There is not a strong lookup relationship between group names and members since we are not doing a "team member registry". Rather, they should be considered parallel properties of groups.
As a concrete example, would you ever write `using group_t = std::map<std::string, std::vector<string>>`?
Also, using dataclasses is not about adapting all API return values, it's about how we pass structured data internally. Dictionaries are perhaps not the best way to represent data if strong type safety guarantees are desired. I will take your advice about using dicts instead of dataclasses for now, but we definitely need to refactor all of these if the data structure becomes more complex in the future.

Reply:
I do not think list+tuples has any advantages over dict here, except for your future expansibility.
I hate over-design and inconsistency. With no extra work, we can represent groups and their members easily. So it should be its current solution.
Plus, this simple tool will not be responsible for dealing with complex data structures inside it in the future. Developers should call these functions and concats complex models by themselves. It is a big mistake to use the same thinking mode in a cpp program.
It makes no sense to create a specific model just for passing a group over internal functions, breaking the consistency over other simple data structures that other functions return. It will be a lot of work to create models just for making the auto-completion works.

Reply:
It is not over-design. It is simply the most natural representation of data. What is the "extra work" you are referring to? Is `group["name"]` more work than `group.keys()[0]`? Again, dictionaries are tuples with names for each element, so using dictionaries simply as 2-tuples makes no sense. You might say it is "simple" but it is anti-intuitive. If you want to remove the field names, using tuples (`"p1team1", ["Alice", "Bob]`) is the way to go, though I see no reduction of complexity by doing this.

Reply:
It is not over-design. It is simply the most natural representation of data. What is the "extra work" you are referring to? Is `group["name"]` more work than `group.keys()[0]`? Again, dictionaries are tuples with names for each element, so using dictionaries simply as 2-tuples makes no sense. You might say it is "simple" but it is anti-intuitive. If you want to remove the field names, using tuples `("p1team1", ["Alice", "Bob"])` is the way to go, though I see no reduction of complexity by doing this.

Reply:
extra work = "create models for each return values"
dict as 2 tuples = fast lookup for members by group name with no extra cost
I think you should write your own intuitive code in your project instead of here.

Reply:
Did you mean `{"p1team1": ["a", "b"], "p1team2": ["c", "d"]}` over `[{"p1team1": ["a", "b"]}, {"p1team2": ["c", "d"]}]`? We weren't understanding each other correctly. I thought you were talking about the latter, which makes no sense anyway.
If you think about it, "models" simply exist. By doing it your way, there is also a "model", it's one that focuses more on finding who is in a specific group while mine focuses more on finding what groups are there. This eventually boils down to personal preferences, and I don't think one design has an advantage over the other.
This isn't a really important issue. If you do think in the foreseeable future this data structure won't be used anywhere else, there is no good reason to reject the entire PR due to some random trivia like this.

Reply:
Yes. I prefer `{"p1team1": ["a", "b"], "p1team2": ["c", "d"]}`. When you explicitly define a model, all things will become a model and it will make things much more complicated. Believe me, I have done so far in horse for a long time.